Subject: Linux-RT 2.6.25.4-RT
From: http://www.kernel.org/pub/linux/kernel/projects/rt/
Acked-by: Sven-Thorsten Dietrich <sdietrich@suse.de>
From lethal@linux-sh.org Fri May 18 06:46:43 2007
Return-Path: <lethal@linux-sh.org>
Received: from smtp.ocgnet.org (smtp.ocgnet.org [64.20.243.3]) by
	mail.tglx.de (Postfix) with ESMTP id 0FCC865C065 for <tglx@linutronix.de>;
	Fri, 18 May 2007 06:46:43 +0200 (CEST)
Received: from smtp.ocgnet.org (localhost [127.0.0.1]) by smtp.ocgnet.org
	(Postfix) with ESMTP id 616355203FB; Thu, 17 May 2007 23:46:39 -0500 (CDT)
X-Spam-Checker-Version: SpamAssassin 3.1.3-gr0 (2006-06-01) on
	smtp.ocgnet.org
X-Spam-Level: 
X-Spam-Status: No, score=0.0 required=5.0 tests=none autolearn=no 
	version=3.1.3-gr0
Received: from master.linux-sh.org (124x34x33x190.ap124.ftth.ucom.ne.jp
	[124.34.33.190]) (using TLSv1 with cipher DHE-RSA-AES256-SHA (256/256
	bits)) (No client certificate requested) by smtp.ocgnet.org (Postfix) with
	ESMTP id E1F585203E0; Thu, 17 May 2007 23:46:38 -0500 (CDT)
Received: from localhost (unknown [127.0.0.1]) by master.linux-sh.org
	(Postfix) with ESMTP id 4984664C7C; Fri, 18 May 2007 04:46:00 +0000 (UTC)
X-Virus-Scanned: amavisd-new at linux-sh.org
Received: from master.linux-sh.org ([127.0.0.1]) by localhost
	(master.linux-sh.org [127.0.0.1]) (amavisd-new, port 10024) with ESMTP id
	BE+H5LV2TYuQ; Fri, 18 May 2007 13:46:00 +0900 (JST)
Received: by master.linux-sh.org (Postfix, from userid 500) id 08A5664C7D;
	Fri, 18 May 2007 13:46:00 +0900 (JST)
Date: Fri, 18 May 2007 13:45:59 +0900
From: Paul Mundt <lethal@linux-sh.org>
To: Ingo Molnar <mingo@elte.hu>, Thomas Gleixner <tglx@linutronix.de>
Cc: linux-kernel@vger.kernel.org
Subject: [PATCH -rt] futex_performance_hack sysctl build fix
Message-ID: <20070518044559.GB22660@linux-sh.org>
Mail-Followup-To: Paul Mundt <lethal@linux-sh.org>, Ingo Molnar
	<mingo@elte.hu>, Thomas Gleixner <tglx@linutronix.de>,
	linux-kernel@vger.kernel.org
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.5.13 (2006-08-11)
X-Virus-Scanned: ClamAV using ClamSMTP
X-Evolution-Source: imap://tglx%40linutronix.de@localhost:8993/
Content-Transfer-Encoding: 8bit

-rt adds a futex_performance_hack sysctl, which is only defined if
kernel/futex.c is built in. This fixes the build in the CONFIG_FUTEX=n
case.

Signed-off-by: Paul Mundt <lethal@linux-sh.org>

--

 kernel/sysctl.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-2.6.25.4-rt4/kernel/sysctl.c
===================================================================
--- linux-2.6.25.4-rt4.orig/kernel/sysctl.c	2008-05-29 09:46:52.000000000 -0400
+++ linux-2.6.25.4-rt4/kernel/sysctl.c	2008-05-29 09:46:52.000000000 -0400
@@ -359,6 +359,7 @@ static struct ctl_table kern_table[] = {
 		.proc_handler	= &proc_dointvec,
 	},
 #endif
+#ifdef CONFIG_FUTEX
 	{
 		.ctl_name	= CTL_UNNUMBERED,
 		.procname	= "futex_performance_hack",
@@ -367,6 +368,7 @@ static struct ctl_table kern_table[] = {
 		.mode		= 0644,
 		.proc_handler	= &proc_dointvec,
 	},
+#endif
 	{
 		.ctl_name	= CTL_UNNUMBERED,
 		.procname	= "prof_pid",
